tests/ashley/test_forum_lti_view.py

```diff
@@ -100,6 +99,100 @@ def test_post_with_an_inactive_user(self):
+            f"/lti/forum/{forum_uuid}",
+            data=urlencode(sign_parameters(passport, lti_parameters, url)),
+            content_type=CONTENT_TYPE,
+            follow=True,
+        )
+
+        # A LTIContext and a Forum should have been created
+        context1 = LTIContext.objects.get(
+            lti_id=context1_id, lti_consumer_id=passport.consumer
+        )
+        forum1 = Forum.objects.get(lti_id=forum_uuid, lti_contexts__id=context1.id)
+        self.assertEqual(LTIContext.objects.count(), lticontext_count + 1)
+        self.assertEqual(Forum.objects.count(), forum_count + 1)
+
+        # The response should be a redirection to the forum URL
+        self.assertRedirects(response, f"/forum/forum/{forum1.slug}-{forum1.id}/")
+
+        # We use the same lti_parameters except the context
+        context2_id = "course-v1:testschool+login+0002"
+        lti_parameters2 = {
+            "user_id": "643f1625-f240-4a5a-b6eb-89b317807963",
+            "lti_message_type": "basic-lti-launch-request",
+            "lti_version": "LTI-1p0",
+            "resource_link_id": "aaa",
+            "context_id": context2_id,
+            "lis_person_contact_email_primary": "ashley@example.com",
+            "lis_person_sourcedid": "testuser",
+            "launch_presentation_locale": "en",
+            "roles": "Instructor",
+        }
+        forum_count = Forum.objects.count()
+        lticontext_count = LTIContext.objects.count()
+        # We request the same LTI launch source url
+        response = self.client.post(
+            f"/lti/forum/{forum_uuid}",
+            data=urlencode(sign_parameters(passport, lti_parameters2, url)),
+            content_type=CONTENT_TYPE,
+            follow=True,
+        )
+        # A new lti context should have been created
+        self.assertEqual(LTIContext.objects.count(), lticontext_count + 1)
+        # A new forum should have been created
+        self.assertEqual(Forum.objects.count(), forum_count + 1)
+        # The response should be a redirection to the forum URL
+        context2 = LTIContext.objects.get(
+            lti_id=context2_id, lti_consumer_id=passport.consumer
+        )
+        forum2 = Forum.objects.get(lti_id=forum_uuid, lti_contexts__id=context2.id)
+        self.assertRedirects(response, f"/forum/forum/{forum2.slug}-{forum2.id}/")
+
+        # we resign the request for context1
+        forum_count = Forum.objects.count()
+        lticontext_count = LTIContext.objects.count()
+        response = self.client.post(
+            f"/lti/forum/{forum_uuid}",
+            data=urlencode(sign_parameters(passport, lti_parameters, url)),
+            content_type=CONTENT_TYPE,
+            follow=True,
+        )
+        # No new lti context should have been created
+        self.assertEqual(LTIContext.objects.count(), lticontext_count)
+        # No new forum should have been created
+        self.assertEqual(Forum.objects.count(), forum_count)
```

Comment:
No need to re-count these each time IMO.
You can rename them `initial_forum_count` and `initial_lticontext_count` if it makes it clearer.

```suggestion
        response = self.client.post(
            f"/lti/forum/{forum_uuid}",
            data=urlencode(sign_parameters(passport, lti_parameters, url)),
            content_type=CONTENT_TYPE,
            follow=True,
        )
        # No new lti context should have been created
        self.assertEqual(LTIContext.objects.count(), lticontext_count + 2)
        # No new forum should have been created
        self.assertEqual(Forum.objects.count(), forum_count + 2)
```

Reply:
I've changed it this way as requested. Thank you for you feedback.